src/node/stage/approved.rs

```diff
@@ -1316,6 +1294,24 @@ impl Approved {
         let _ = self.shared_state.update_member(member_info, proof);
     }
 
+    async fn handle_send_message_event(
+        &mut self,
+        message: PlainMessage,
+        proof_chain: SectionProofChain,
+        proof: Proof,
+    ) -> Result<()> {
+        let message = Message::section_src(
+            message.src,
+            proof.signature,
+            message.dst,
+            message.variant,
+            proof_chain,
```

Comment:
just to confirm, that when we carry out aggregator on this vote, is there any check to be carried out to verify the proofshare is from the proof_chain ?
it seems the `Message::section_src` doesn't carry out any check whether the proof is from the proof_chain ?

also, wondering shall we also carry out a check that the `proof_chain` is a valid chain? i.e. at least contains the knowledge that we have of the srouce section?